doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst

```diff
@@ -0,0 +1,300 @@
+       | key_bits: 255
+     - 2
+     - No
+     - Yes
+     - Yes
+   * - ED25519 public key
+     - | key_type: PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_TWISTED_EDWARDS)
+       | key bits: 255
+     - 2
+     - No
+     - Yes
+     - Yes
+.. [1] Keys with the Encrypted usage scheme (CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED) will require two additional KMU slots to store the nonce and the authentication tag.
+
+Storing keys in KMU
+-------------------
+
+Applications can store keys in KMU slots using the standard PSA cryptographic key management operations psa_import_key, psa_generate_key or psa_copy_key.
+Additionally, KMU slots can be provisioned using the nrfutil utility. (TODO: Link to nrfutil docs when ready)
+
+.. note::
+   If there is a power failure when provisioning a key with persistence PSA_KEY_PERSISTENCE_READ_ONLY or CRACEN_KEY_PERSISTENCE_REVOKABLE, it might not be possible to recover the key slot.
+   Provisioning of read-only keys should be restricted to controlled environments (production).
+
+The following special error codes are available:
+
+* PSA_ERROR_ALREADY_EXIST: One of the required key slots has already been provisioned
+* PSA_ERROR_NOT_SUPPORTED: Unsupported key type
+
+Removing or revoking keys from the KMU
+--------------------------------------
+
+Keys are deleted or revoked using the psa_destroy_key function.
+Calling the psa_destroy_key function on keys that have the persistence CRACEN_KEY_PERSISTENCE_REVOKABLE, will mark the associated KMU slots as revoked, preventing the slots from being re-used for new keys.
+
+
+Using KMU keys
+--------------
+
+Keys stored in the KMU can be used in standard PSA cryptographic operations for encryption, decryption, signing a hash or a message, and verifying a hash or a message, given that the corresponding PSA_KEY_USAGE_* flags are set.
+
+KMU push operations
+
+Keys with the usage schemes CRACEN_KMU_KEY_USAGE_SCHEME_PROTECTED and CRACEN_KMU_KEY_USAGE_SCHEME_SEED can push data from the slots directly to CRACEN registers that are not accessible by the CPU.
+These usage schemes are supported for pushing symmetric keys used for cipher operations, and for seeds for use by the CRACEN IKG.
+
+Keys with the usage scheme CRACEN_KMU_KEY_USAGE_SCHEME_RAW, such as symmetric keys, must be temporarily pushed to a RAM location, before CRACEN can load them into the asymmetric engine before running operations like sign or verify.
+Key slots that are tagged with CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED also have to be decrypted to a temporary push location in RAM before they are used by CRACEN.
+
+When the application is built with TF-M, this temporary push location is protected inside the secure processing environment to avoid exposing the key material to the non-secure application.
+If TF-M is not used, the keys are pushed to a reserved RAM area at location 0x20000000-0x20000064 (kmu_push_area).
+
+The following special error codes are available:
```

Comment:
The PSA Crypto API list error codes per function. Do these error codes come from one specific function or from any PSA Crypto API function?